src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,172 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size()) {
+            throw new IndexOutOfBoundsException();
+        }
```

Comment:
move it to a separate method, this is a code duplication
```suggestion
        if (index < 0 || index > size) {
            throw new IndexOutOfBoundsException();
        }
```